server/internal/graphql/resolver/meshsync.go

```diff
@@ -218,7 +218,28 @@ func (r *Resolver) resyncCluster(ctx context.Context, provider models.Provider,
 			if provider.GetGenericPersister() == nil {
 				return "", ErrEmptyHandler
 			}
-			err := provider.GetGenericPersister().Where("cluster_id = ?", sid).Delete(&meshsyncmodel.Object{}).Error
+
+			err := provider.GetGenericPersister().Where("id IN (?)", provider.GetGenericPersister().Table("objects").Select("id").Where("cluster_id=?", sid)).Delete(&meshsyncmodel.KeyValue{}).Error
```

Comment:
Why to manually do this? Does the deletion in objects table alone not trigger the removal of rows from other tables with foreign key reference provided the gorm tags that we have?

Reply:
No, the deletion of `objects` table alone doesn't trigger deletion of rows in other tables with same `id`

Reply:
Changing `gorm` tag for `ONDELETE` to `cascade` should work. 

https://github.com/meshery/meshsync/blob/6f4eefabd09a9b8f0e8aa0919a00607a988141a5/pkg/model/model.go#L17

Reply:
@MUzairS15  I was trying it but there seems to be some problem as just adding that constraint didn't work. 

So currently I'm explicitly deleting the rows from other tables by matching their foreign key.
I would revisit this area while implementing the tombstone feature as that too requires a change in DB schema.